core/lib/web3_decl/src/namespaces/en.rs

```diff
@@ -20,4 +20,11 @@ pub trait EnNamespace {
         block_number: MiniblockNumber,
         include_transactions: bool,
     ) -> RpcResult<Option<SyncBlock>>;
+
+    /// Lists all tokens created at or before the specified `block_number`.
+    ///
+    /// This method is used by EN after snapshot recovery in order to recover token records.
+    #[method(name = "syncTokens")]
+    async fn sync_tokens(&self, block_number: Option<MiniblockNumber>)
+        -> RpcResult<Vec<TokenInfo>>;
```

Comment:
As an alternative, this method could be placed in the `zks` namespace as `getAllTokens`. It depends on whether we want to make this method (eventually) public / usable for use cases beyond snapshot recovery.